docs/tests/visdiff.js

```diff
@@ -0,0 +1,188 @@
+const puppeteer = require("puppeteer");
+const expect = require("chai").expect;
+const path = require("path");
+const serve = require("serve");
+const fs = require("fs");
+const PNG = require("pngjs").PNG;
+const pixelmatch = require("pixelmatch");
+const base64Img = require('base64-img');
+
+const testDir = "tests";
+const goldenDir = "tests/golden";
+
+describe("docs site", () => {
+  let server, browser, page;
+  before(async () => {
+    const serveDir = path.join(__dirname, "../build");
+    server = serve(serveDir, {
+      port: 8080,
+      ignore: ["node_modules"]
+    });
+    // And its wide screen/small screen subdirectories.
+    if (!fs.existsSync(`${testDir}/wide`)) fs.mkdirSync(`${testDir}/wide`);
+    if (!fs.existsSync(`${testDir}/wide/component`)) {
+      fs.mkdirSync(`${testDir}/wide/component`);
+    }
+    browser = await puppeteer.launch();
+    page = await browser.newPage();
+  });
+
+  after(() => {
+    server.stop();
```

Comment:
Add `browser.close()` here, then `--exit` on `mocha` in `package.json` shouldn't be required (but still have to test that one out).

Reply:
oh TIL